src/StatusQ/Controls/StatusTagSelector.qml

```diff
@@ -0,0 +1,109 @@
+            spacing: 8
+            StatusBaseText {
+                Layout.preferredWidth: 22
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+                color: Theme.palette.baseColor1
+                text: "To: "
+            }
+
+            ListView {
+                id: namesList
+                //TODO check if 220 is the correct limit for width
+                Layout.preferredWidth: (childrenRect.width > 220) ? 220 : childrenRect.width
+                implicitHeight: 30
+                visible: (count > 0)
+                Layout.alignment: Qt.AlignVCenter
+                model: namesModel
+                orientation: ListView.Horizontal
+                spacing: 8
+                clip: true
+                onContentWidthChanged: {
+                    positionViewAtEnd();
+                }
+
+                delegate: Rectangle {
+                    id: nameDelegate
+                    width: (nameText.contentWidth + 40)
+                    height: 30
+                    color: Theme.palette.primaryColor1
+                    radius: 8
+                    StatusBaseText {
+                        id: nameText
+                        anchors.left: parent.left
+                        anchors.leftMargin: 8
+                        anchors.verticalCenter: parent.verticalCenter
+                        color: Theme.palette.white
+                        text: name
+                    }
+                    ToolButton {
+                        width: 24
+                        height: 24
+                        anchors.verticalCenter: parent.verticalCenter
+                        anchors.left: nameText.right
+                        background: null
+                        contentItem: StatusIcon {
+                            color: Theme.palette.white
+                            icon: "close"
+                        }
+                        onClicked: {
+                            namesModel.remove(index, 1);
+                        }
+                    }
+                }
+            }
+
+            TextEdit {
+                id: edit
+                Layout.fillWidth: true
+                Layout.preferredHeight: 44
+                verticalAlignment: Text.AlignVCenter
+                visible: (namesModel.count < 5)
+                enabled: visible
+                focus: true
+                font.pixelSize: 15
+                font.family: Theme.palette.baseFont.name
+                color: Theme.palette.directColor1
+                Component.onCompleted: {
+                    forceActiveFocus();
+                }
+            }
+
+            StatusBaseText {
+                visible: (namesModel.count === 5)
+                Layout.preferredWidth: 120
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignRight
+                font.pixelSize: 10
+                color: Theme.palette.dangerColor1
+                text: "5 USER LIMIT REACHED"
```

Comment:
This is a tag selector component that doesn't necessarily know the semantics of "users".

So few thoughts:

1. This message shouldn't be hardcoded (needs to be i18n'able anyways)
2. Would be cool if this component would leverage `StatusValidator` and then this message would be retrieved from the validator instance

Do you think it's feasible to replace `TextEdit` with `StatusBaseInput` (and tweak it as needed) so we get validation capabilities?

Ultimately, all (custom) controls should have validation built-in following the same APIs

Reply:
Thought about it (didn't try though), the problem is the internal Rectangle it has that paints the grey background and that's not customizable, unless I hide it, but still the red borders and the whole "invalid state" UI will look weird. Other than this, I don't see any invalid state [here](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba⎜Desktop?node-id=391%3A100984)

Reply:
done

Reply:
Okay, that's fair for now!